src/ImageSharp/Formats/Png/PngDecoderCore.cs

```diff
@@ -971,6 +977,94 @@ private void ReadCompressedTextChunk(PngMetadata metadata, ReadOnlySpan<byte> da
             {
                 metadata.TextData.Add(new PngTextData(name, uncompressed, string.Empty, string.Empty));
             }
+
+            if (name.Equals("Raw profile type exif", StringComparison.OrdinalIgnoreCase))
+            {
+                this.ReadLegacyExifTextChunk(baseMetadata, uncompressed);
+            }
+        }
+
+        /// <summary>
+        /// Reads exif data encoded into a text chunk with the name "raw profile type exif".
+        /// This method was used by ImageMagick, exiftool, exiv2, digiKam, etc, before the
+        /// 2017 update to png that allowed a true exif chunk.
+        /// </summary>
+        /// <param name="metadata">The <see cref="ImageMetadata"/> to store the decoded exif tags into.</param>
+        /// <param name="data">The contents of the "raw profile type exif" text chunk.</param>
+        private void ReadLegacyExifTextChunk(ImageMetadata metadata, string data)
+        {
+            ReadOnlySpan<char> dataSpan = data.AsSpan();
+            dataSpan = dataSpan.TrimStart();
+
+            if (!dataSpan.Slice(0, 4).ToString().Equals("exif", StringComparison.OrdinalIgnoreCase))
+            {
+                // "exif" identifier is missing from the beginning of the text chunk
+                return;
+            }
+
+            // Skip to the data length
+            dataSpan = dataSpan.Slice(4).TrimStart();
+            int dataLengthEnd = dataSpan.IndexOf('\n');
+            int dataLength = int.Parse(dataSpan.Slice(0, dataSpan.IndexOf('\n')).ToString());
```

Comment:
```suggestion
            int dataLength = int.Parse(dataSpan.Slice(0, dataSpan.IndexOf('\n')));
```
`int.Parse` has an overload for ROS<char>, so the intermediate string-allocation can be avoided.

Reply:
It's [missing](https://apisof.net/catalog/71bfcfac-9ac2-5416-bb5c-b5c645090d9d) in some of our target frameworks. Unfortunately this looks like a candidate for code like this:

```C#
internal static int ParseInt32(ReadOnlySpan<char> span)
{
#if NETSTANDARD2_1 || NETCOREAPP2_1_OR_GREATER
    return int.Parse(span, CultureInfo.InvariantCulture);
#else
    return int.Parse(span.ToString(), CultureInfo.InvariantCulture);
#endif
}
```

Reply:
@antonfirsov Is there some "Utility" static class etc where you would like that added? Or just stick it in `PngDecoderCore` for now?

Reply:
Yes, I don't see other use cases for it ATM.

Reply:
Very well. A similar function, `StringEquals` for comparing two `ReadOnlySpan<char>`s, was also added to `PngDecoderCore`.